### ISSUE TITLE
Bio Weekly - Home Mouthwash Collection Metrics

### ISSUE THREAD

**Issue by HopkinsDC** [Analyticsphere/metricsReportsRequests#88]
Please generate the six new tables and one new figure for the home mouthwash collection metrics. These new metrics will be added to the weekly biospecimen metrics report following Table 26; the mock-up tables are numbered to reflect this. 

**Requestor(s):** Domonique

**Table or Figure title:** See mock-up for titles.

**Metrics to be updated or created:** Create new metrics

**Variables needed (include variable name and Concept ID):** 
- RcrtES_Site_v1r0 (827220437) – Sites 
- BioSpm_Setting_v1r0 (650516960) – Collection setting
- BioRec_CollectFinal_v1r0 (410912345) - Complete Collection
- BioRec_CollectFinalTime_v1r0 (556788178) – Collection date/time 
- BioFin_BaseMouthCol_v1r0 (684635302) – Baseline mouthwash collection
- BioKit_KitType_v1r0 (379252329) – Kit Type
- BioKit_KitStatus_v1r0 (221592017) – Kit status
- BioKit_KitShipTm_v1r0  (661940160) – Date/time kit shipped
- BioKit_KitRecdTm_v1r0 (826941471) – Date/time kit received
- BioKit_CollCardRet_v1r0 (137401245) – Collection card returned flag 
- (Source Question) Select Package Condition - Home Collections (633640710)

**Specify numerator and denominator by variable name where applicable:** See mock-up for denominator and numerator for each table (below each table in bold)

**Mock-up table if applicable:** 
[Mock-Up_HC-MW Metrics_102323.docx](https://github.com/Analyticsphere/metricsReportsRequests/files/13300137/Mock-Up_HC-MW.Metrics_102323.docx)

**Notes:** N/A

**Approved by:**

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  Quick question, without data I can try writing a code for these tables but won't know if they worked until I have actual data. I may try this in stage if we have home collection data there for these variables. Are we still on the DEV phase?

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 Yes, we are still in the dev phase. We will be testing home collections in dev today and tomorrow, so there should be data in dev to test the code on next week, but until then, you're fine to just draft the code for now. I'll add a comment here when we finish our initial testing in dev so you do a test run.

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  Ok Thanks! The structure of DEV won't work for the code but I'll just draft the code for now like you said

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  For Table 28, should kit_status="Pending" be a column as well, or should I exclude those or put them in with no status?

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 Kit status is present in both the Kit Table and the Participants Table. "Pending" will only exist in the Kit Table for kits that have been created but have not yet been linked to a participant. "Address Printed" only exists in the Participants Table for when the participant's address is printed in the BPTL dashboard; the kit status is NULL until then. Once linked, the kit status in both tables is updated to "Assign" and will update to "Shipped" and "Received" simultaneously as these events occur. 

Here's the home mouthwash collection data structure for your reference: https://nih.app.box.com/file/1306274192476

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC Ok so there shouldn't be any participants in Table 28 that should have a status of pending? Or do you want that column added?

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 No, don't add a Pending column; participants will not have a Pending status.

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  Got it, thanks. This code should be ready to test once we have data in prod

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 Hi Kelsey, home collections will be in stage on 11/27 or 11/28. Can the home collection metrics code be tested in stage?

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  If all the variables needed for the tables are in the biospecimen or participants table in BQ, then yes.

Right now these variables aren't in there (which may make sense for now, but would just need to be answered to test)
-137401245
-633640710
-950521660
-545319575
-938338155
-205954477
-289239334
-992420392
-541085383

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 A new table was created for the home collections (the Kit Table), which is where the variables listed above, and other variables for the tables, are. These Kit Table variables will have values in stage after we test on 11/28.

Will you be able to test these at that time?

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  Oh that's right, I saw Nicole mention the Kit table to Jake. As long as that table has been flattened in stage and those questions answers, yes I would be able to test these at that time

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 Awesome, thank you! I'll follow-up here when the stage testing is done so you can test.

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 Testing in stage is complete for the most part, so you can try out your home collection metrics code now.

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  d_143615646_d_826941471 hasn't been answered yet so I can't test Table 30 or Table 31. Of the four participants in stage where d_173836415_d_266600170_d_8583443674_d_221592017==375535639 (kit status received) none have saved conditions so I can't run Table 32 yet 


![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/b0fcf456-78fe-45fb-83ad-4aba75ec42a8)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/986d1cbc-cff9-42a1-951e-1d36b92823d5)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/297b15dc-8f37-4648-844e-018144a1463a)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/ec120668-1973-4487-aa58-6db77b2bb436)

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 The tables you were able to run look fine except for Table 27. The columns are mislabeled; can you update those to the labels I have in the mock-up? Also, I think I used the wrong CID for Table 27's numerator. Please update the table using the following denominator and numerator below:

Denominator = Complete clinical collections (650516960 = 1; 410912345 = 1)
Numerator = Mouthwash collected in home setting (**915179629** = 2 and 684635302 = 1) - If 915179629 or 684635302 is NULL, the record should be included in the "No Home Mouthwash Collected" column count.

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  Thank you for pointing that out! There was a filter missing in Table 27. But I think the denominator would need to be 650516960 = 2 (Collection setting=home) instead of 650516960 = 2 (Collection setting=research) right?

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 It'll be better to look at the mouthwash collection setting specifically (BioSpm_MWSettingBL_v1r0 - 915179629=2 [Home]) for Table 27's denominator.

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  Got it, I misunderstood some of the requirements but here's the update Table 27:
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/24848761-93a9-416f-887d-dee91f355da6)


Tables 30-32 are on hold- d_143615646_d_826941471 hasn't been answered yet so I can't test Table 30 or Table 31. Of the four participants in stage where d_173836415_d_266600170_d_8583443674_d_221592017==375535639 (kit status received) none have saved conditions so I can't run Table 32 yet

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 None of the kits in stage have a kit received date populated?

Can you test the code again with the NCI site added too (RcrtES_Site_v1r0 [827220437] = 9 (National Cancer Institute [517700004])?

**Comment by KELSEYDOWLING7** [comment]
So the problem is that we have 8 values for d_143615646_d_826941471, but for all 8 d_410912345 (BioRec_CollectFinal_v1r0) is nulll.... and that's a filter when I first pull the data in from BQ.

As for table 32 that was my mistake, the responses aren't nested the way I thought they would be so the code was off
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/249c162c-e398-4567-ade1-3803f27a107f)

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 Table 32 looks good, but can you add a "Total" row to it?

That's weird that there's a filter for stage data; can you remove it, or is that filter for all data pulled from BQ?

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  Yes, here's the updated table:
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/230f4c1b-3669-4835-9bff-b6f68e0b3dd6)


And that's the filter we decided to add to all the biospecimen data a while back, so I'd have to do a separate BQ pull if we want only this data to include non-finalized collections

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 The row titles are showing as 1, 2, 3, and 4 now instead of the condition names; can you fix that?

Okay, in that case, don't worry about testing Tables 30 and 31 in stage. You can just have the code ready for when we have production data and check them then; we'll make adjustments if needed. Thanks!

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  Oops! Yes, here's that corrected 

![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/fca84945-dae3-44ff-b997-8bc6bbbb8818)

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 Hi Kelsey, please make the following changes to the home mouthwash collection metrics:

Table 27
- Change title "Table 27. Baseline Home Mouthwash Collection Status Among Clinical Collections by Site" to --> **Table 7.1: Baseline Home Mouthwash Collection Status Among Eligible Participants** 
- Change the last column's label to "Total Eligible Participants"
- Add the following footnote --> **Note: “Eligible participants” are those who had a clinical collection with blood or urine collected on or after January 1, 2024, and did not refuse, withdraw, or pass away prior to January 2024. Home mouthwash collections were launched for eligible participants in January 2024. Home mouthwash collections for backlog collections (before January 2024) will be launched at a later date.** 
- Updated the denominator of this table to be the same as Table 28 [Table 7.2] below.


Table 28
- Change title "Table 28. Home Mouthwash Collection Kit Status Among Eligible Participants by Site" to --> **Table 7.2: Baseline Home Mouthwash Kit Status Among Eligible Participants** 
- Add the following footnote --> **Note: “Eligible participants” are those who had a clinical collection with blood or urine collected on or after January 1, 2024, and did not refuse, withdraw, or pass away prior to January 2024. Home mouthwash collections were launched for eligible participants in January 2024. Home mouthwash collections for backlog collections (before January 2024) will be launched at a later date.** 
- Updated the denominator of the table to be 


Table 29
- Change title "Table 29. Collection Card Status Among Retuned Home Mouthwash Collection Kits by Site" to --> **Table 7.3: Collection Card Status Among Received Baseline Home Mouthwash Kits** 


Table 30
- Change title "Table 30. Time (in Hours) from Home Mouthwash Collection Kit Shipment to BPTL Receipt by Site " to --> **Table 7.4: Time (in Days) from Baseline Home Mouthwash Kit Shipment to BPTL Receipt** 
- Update table calculation to be **in Days**, not in hours


Table 31
- Change title "Table 31. Time (in Hours) from Clinical Biospecimen Collection to BPTL Home Mouthwash Collection Kit Shipment by Site " to --> **Table 7.5: Time (in Days) from Baseline Clinical Collection to Home Mouthwash Kit Shipment** 
- Update table calculation to be **in Days**, not in hours
- Change the collection date/time variable used in the calculation to BioCol_ColTime_v1r0 (678166505)


Table 32
- Change title "Table 32. Home Mouthwash Collection Kit Condition" to --> **Table 7.6: Kit Conditions Among Received Baseline Home Mouthwash Kits** 


Figure 5
- Change title "Figure 5. Weekly Shipped and Received Home Mouthwash Collection Kits" to --> **Figure 7.1: Weekly Shipped and Received Baseline Home Mouthwash Kits** 
- Change the y-axis title to "Number of Kits"
- Replace the underscore with a space in the legend title "Kit_Status"

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  For table 27, should the date of collection be BioCol_ColTime_v1r0 (678166505)? 


Also before we had Table 27's denominator as BioSpm_Setting_v1r0=Clinical and Table 28's denominator as BioKit_KitTypeBL_v1r0=Mouthwash. 

Should those components both be kept, and then these added onto both: HdWd_Activepart_v1r0=no, HdWd_WdConsent_v1r0=no, BioCol_ColTime_v1r0>1/1/24, and (HdWd_Deceased_v1r0=no OR HdWd_DeceasedDt_v1r0>1/1/24)?

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 Yes, please use BioCol_ColTime_v1r0 (678166505) for the date of collection, and yes, keep both and add the other variables you listed.

**Comment by KELSEYDOWLING7** [comment]
@HopkinsDC  Ok thanks. Unfortunately we have 0 participants that fit that description in Stage so I'll have to wait for more data

**Comment by HopkinsDC** [comment]
@KELSEYDOWLING7 That's fine. I doubt that part was included in the stage testing for home collections anyway. Thank you!

**Comment by StephanieWeinstein** [comment]
@KELSEYDOWLING7 , Michelle and I will review this so you can hold for a bit on this

**Comment by erincschwartz** [comment]
@KELSEYDOWLING7 We are looking at the Weekly Biospecimen Metrics Report. Two home mouthwash collection kits were shipped on Friday, we expected to see the Home Mouthwash data in the report today. None have been received yet, but we should see the sent status for the first two kits. Do you know why this isn't showing up?


Also, home mouthwash collection was delayed please adjust the footnotes for Table 7.1 and 7.2 to reflect April launch (not January): 

Note: “Eligible participants” are those who had a clinical collection with blood or urine collected on or after April 1, 2024, and did not refuse, withdraw, or pass away prior to April 2024. Home mouthwash collections were launched for eligible participants in April 2024. Home mouthwash collections for backlog collections (before April 2024) will be launched at a later date.

Also, please update titles for Table 7.3-7.6 and Figure 7.1 to reflect new title convention Baseline-[rest of title]

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  Hi Erin, yes I haven't put any of the MW code into the report yet because not all the variables were in prod on Friday. If any were missing it would cause the whole report to crash and it looks like we're still waiting on many of them to be populated. I believe Michelle mentioned on the Punch List call that we should be getting about 20 kits back tomorrow. That should hopefully give us a majority of them if not all.

As far as table 7.1 and 7.2 and figure 7.2 are you referring to these on the markup?
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/e6b95957-2ed9-42e7-9440-6e077e070ce3)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/849c32b4-eecb-4a53-a15b-91a3462fd1b9)

If so then yes I will update those now with the footnotes and title change. Once we have all the variables I'll rerun the report and post that section here for aproval before updating the automation. If we have all the variables by the end of the week, do you need me to update the report this Friday for next Monday's report as opposed to the next automation update scheduled for Apritl 19th for the April 22nd report?

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  I also notified Jake that we now have data in the kitassembly and mouthwash tables so he can flatten those, as those are both needed for these tables

**Comment by erincschwartz** [comment]
@KELSEYDOWLING7 Yes, I was referring to Table 27 & 28 shown above, just used the new naming convention. To clarify, 2 kits were sent Friday; we expect an additional 18 kits to be sent today, for total of 20 sent out. We do not know how quickly any of the kits will be returned. That said, if we have kits returned and all the data available, can you push it through this Friday? If it is a huge burden to do this out of cycle, I can talk to Bio Team about waiting. I know we are anxious to start seeing the data!

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  Yes, if we have all the variables needed then I can talk to Jake about updating the automation on Friday, or running it manually on Monday

**Comment by erincschwartz** [comment]
> [@KELSEYDOWLING7](https://github.com/KELSEYDOWLING7) Kit status is present in both the Kit Table and the Participants Table. "Pending" will only exist in the Kit Table for kits that have been created but have not yet been linked to a participant. "Address Printed" only exists in the Participants Table for when the participant's address is printed in the BPTL dashboard; the kit status is NULL until then. Once linked, the kit status in both tables is updated to "Assign" and will update to "Shipped" and "Received" simultaneously as these events occur.
> 
> Here's the home mouthwash collection data structure for your reference: https://nih.app.box.com/file/1306274192476

@KELSEYDOWLING7 Just pointing out that Madhuri updated the home mouthwash collection data structure document, linked above. The only variable she updated was "BioKit_Mouthwash" which I do not believe your reports are using but wanted to let you know in case.

**Comment by KELSEYDOWLING7** [comment]
A few of the variables I need did have a nesting structure change so thank you for confirming!

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  It's still too early for a majority of these tables and Figures to run.

So for the first, second, and third HMW, that collection we did receive back but this variable BioCol_ColTime_v1r0 is null for that person so those tables won't run. Joe's email mentioned that he can see a value for that variable, but the value wasn't entered correctly. It's a date instead of a timestamp I'm assuming that's why BQ isn't handling it correctly. 

@jacobmpeters this variable d_678166505 has two rows with different values for the same participant  in the raw table, one that's null and one is a date variable instead of datetime. Can we flatten that as is and add T00:00:00.000Z or does that have to be fixed upstream?

BioKit_KitRecdTmBL_v1r0 is null for everyone so the fourth and fifth table won't run. 

None of the package conditions were filled out so the sixth table won't run. 

BioKit_CollCardMissing_v1r0 is null so the last table won't run.

**Comment by jacobmpeters** [comment]
@KELSEYDOWLING7 Do you mind showing me this issue with `d_678166505` tomorrow? I'm not sure I quite understand but I'm happy to help..

**Comment by KELSEYDOWLING7** [comment]
@jacobmpeters  Of course!

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  A few tables were able to be run this morning! Going to be working with Jake on the duplicate rows for the BioCol_ColTime_v1r0 that is throwing some of the numbers off. But let me know if anything else stands out that needs to be changed. I updated the footnotes for Table 7.1, Table 7.2, and Figure 7.1 as requested above. 

![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/a3826086-ca29-4874-8a13-84623ee1fb92)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/ad7aa0f7-99c4-4231-8a25-b404a91de0fb)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/d43aa49d-34d6-4a54-a9f5-e4057497bccd)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/fdba19d7-9d30-4c32-9a30-8f955942e443)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/247e475c-28c5-448e-9d5a-dbf1575f2cc9)

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  Here are the updates removing the duplicate row issues! Please let me know if you need the dates on the x-axis brought to a more recent date to avoid that gap

![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/b686a4f1-1eb1-4c1f-b611-76e59059b0bb)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/8a2619ef-b44b-47e8-b54f-b84bbf5a3950)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/84d52cd3-30dc-4cd5-865d-50b1bb5c0e4e)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/6172cb03-f085-4b5f-93df-b19e137bbd3c)

**Comment by erincschwartz** [comment]
@KELSEYDOWLING7 Please see notes on the updates we discussed on our call. I received confirmation on Table 7.5 from Bio Team, note included below:

**Table 7.1** - Update title to: Baseline- Home Mouthwash Kit Status Among Eligible Participants 
		- Change columns to:    Total Eligible-- Total Kits Shipped -- Total Kits Received  
**Table 7.2** - Remove this table
**Table 7.3**-  Update title: Baseline- Collection Card Status Among Home Mouthwash Kits Received
**Table 7.4** - Add  "N" between site name and min 
**Table 7.5** - Add  "N" between site name and min 
	       - Clinical Collection is date of clinical blood or urine collection, whichever occurred first
**Table 7.7** - Update title to:  Baseline- Mouthwash Biospecimen Survey Status Among Mouthwash Kits in 'Shipped' Status
			- Update "Total Shipped Kits" column header to "Total Kits in 'Shipped' Status" 
**Figure 7.1** Update start date on X axis
			- Make sure this is including all kits ever shipped that week, not just those in 'shipped' status
**Table 7.9**- Change title to: Baseline- Mouthwash Biospecimen Survey Completion Time Relative to Collection Time Reported on Collection Card
		    - Change Before Collection column header to:  "1+ Day(s) before Collection"
		    - Add footer-  Note: Only includes data from mouthwash kits returned with completed Collection Card

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  For consistency it may make more sense for Table 7.1 columns to be total shipped, total received, and THEN total eligible. In all the other tables we have the column with the denominator on the far right as reference, unless we want eligible to mean eligible but not yet shipped or received. Also for that table I would need the conditionality for what makes a participant eligible.

**Comment by KELSEYDOWLING7** [comment]
Also do you know if something like this might be acceptable for Figure 7.1? With maybe another footnote that 'Total Shipped' is the weekly total shipped, including those received.

![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/73435cd4-c206-470c-84ea-b7c899515079)

**Comment by erincschwartz** [comment]
@KELSEYDOWLING7 Confirming my understanding, is the dashed line a cumulative total for all weeks?

**Comment by KELSEYDOWLING7** [comment]
The dashed is the weekly total of Shipped (Shipped+Received). Although now that I write that, that assumes that the receiving date and shipped date are the same week each week.... this may be more complicated then I thought. Please disregard the figure.

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  Sorry for all the back and forth! I take that back. I was able to use the weekly shipped date counts as we discussed to verify the "Total Shipped" is the total number of kits shipped per week, regardless of whether they were received that week. I overlaid that line on the existing plot that just looks at counts of Kit Statuses and the plot above is correct, if that style is acceptable. 

I can also remove the "Shipped" counts if its confusing. I'm struggling to get the "Total Shipped" line to also have the dots on the week, which is why I left "Shipped" on there for now as a reference. Please let me know if that makes sense or if you have any questions

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  Last question I swear! The way the table was most recently requested would look something like this, with all eligible, counts of total ever shipped per site and counts of ever received per site. I think it may be a bit confusing against our other tables as a participant can be in all counted three columns and the denominator is on the far left.
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/369d510a-3fbd-4c6c-9efe-615132d33b13)

Alternatively we could have something like this that matches the title, with just a cross table of site and kit status among all eligible:
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/20cc885d-6eed-4250-967b-0ce0394cb870)

**Comment by erincschwartz** [comment]
@KELSEYDOWLING7  we will go with Figure 7.1 as shown above., with total shipped, shipped and received. No problem that total shipped does not have the dot. 

For Table 7.1, we like your alternative suggestion above. Please update the order of the columns though: 
No Status > Address Printed >Shipped> Received > Total 

Thanks!

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  Thanks! I'll update the title numbers now that Table 7.2 is removed, but I didn't want to make it confusing for this review. Also, the current conditionality for Figure 7.1 (kit type=mouthwash) is preventing any counts of address printed from showing. Not sure if that's a problem

![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/a9ac0bae-ee0f-431b-90ff-4914061cb9df)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/282c88ac-85a1-49d0-b9e4-d88d2d9a565e)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/c7e4ea2d-531a-4cdd-b47f-d73d8c756fcc)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/ab860736-c0f9-4b64-aa25-228c472e0563)

**Comment by erincschwartz** [comment]
@KELSEYDOWLING7 Table 7.5 does not look like it is capturing the right data, all clinical collection sites should be included since kits have been shipped to participants at all sites and the N is not correct for the 3 sites included. This is supposed to show the time in days since the clinical blood/urine collection (whichever occurred first) to the date of kit shipment. Can you look at this?

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  You're right, thank you for pointing that out. There was a typo in one of the table merges. Please let me know how this looks.
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/954d4618-9bc8-4328-8e14-c8c34a664ba0)

**Comment by KELSEYDOWLING7** [comment]
Two participants aren't counted in that table that did have a MW kit shipped to them because their first blood or urine collection time was after kit shipment (same day but later that day): Connect IDs 3762092872 and 9055416970. Please let me know if you want the table updated to include times of 0

**Comment by erincschwartz** [comment]
@KELSEYDOWLING7 I will take the comment above back to the Bio team for those two cases, Connect IDs 3762092872 and 9055416970----the sort order for printing labels/assigning kits was just updated in the Prod push, so kits will likely not ever be shipped the same day as blood/urine collection. 

Table 7.6 is showing that 5 kits were receipted with "no conditions selected," however, we just confirmed in the Dash that FNL must select package condition before receipting the kit in the Dash. Can you look into this? 

Also, still working to perfect Figure 7.1----seeing more data in there now makes us think we do not actually want the "shipped" count showing. We just want to see absolute values: how many kits were shipped out from FNL in a given week and how many kits in total were received in a given week. With removal of this line, we can convert the dashed line for Total Shipped into a solid line, yellow works. 

Please clarify your note above: _Also, the current conditionality for Figure 7.1 (kit type=mouthwash) is preventing any counts of address printed from showing. Not sure if that's a problem._ Address printed should not factor into this Figure, so I am not sure what this means. 

Thanks!

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  7.6 was linked to the delayed kit_assembly flattening. That's now fixed. 
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/a7862927-4796-4b96-874f-852055f61a6c)


Table 7.1- shipped status removed. If counts for address printed aren't needed here, please disregard the comment. Please also disregard the b in the table title, I was playing around with different versions
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/cd8fe295-f188-4a18-9b59-662e26ac5677)

**Comment by erincschwartz** [comment]
@KELSEYDOWLING7 Thanks for fixing 7.6! 

Figure 7.1 looks off now, the first week 2 were shipped, the second week 18 were shipped, third week 100 were shipped. Also, forgot to mention earlier, can we start at 3/25, to get rid of even more blank space along the x-axis? And, can you eliminate showing the data for the current week, since it is not complete? And, if run on Monday morning, will likely always show a plummet to 0?

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  I'll look into the counts for 7.1, specifically for 4/1, and have updated the x axis. The dates are counting the Monday prior until the day before, which when run, would be Sunday so there will always be full counts. For example Monday will have 4/29/24 on the x-axis but will have the full count of 4/22-4/28 and would never plummet to 0 unless there were no counts or if it was run mid-week before anything was shipped.

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz 
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/1b9fa5fe-fe21-48c0-aff0-52c8971c2ac6)

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  As requested here are the tables refreshed with this weeks data. The plot is pending adjustment to Monday collections week assignments.
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/1741b5a0-ce17-49d0-85df-b4d89996494d)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/8355ece0-fd09-477f-9633-3a578542dbc1)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/037afca7-f175-4abb-8f80-ec3cbb759d14)

**Comment by erincschwartz** [comment]
@KELSEYDOWLING7  Thank you!

For Table 7.9, please add columns for:  2 Days before Collection, 3 Days before Collection, and 4+ Days before Collection

We were originally concerned about people completing the survey much after the collection but now we realize we need more insight into numbers of participants who are completing it well before actually completing the collection.

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  Sure thing! I added 1 day before collection as well to account for all scenarios
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/46439e3e-4410-4f11-94e4-a91dfebb4b28)

**Comment by StephanieWeinstein** [comment]
Hi Kelsey, 
Thank you, that is very helpful.  It is showing us that a large percentage of participants are filling out the mouthwash survey too soon, which is something we need to address.  
Thank you!
Stephanie

**Comment by erincschwartz** [comment]
@KELSEYDOWLING7 We noticed a discrepancy in the KPCO data. In table 7.1 above, it shows that 5 kits have been returned; Tables 7.3 and 7.4 seem to show that 7 kits have been returned. Can you look into this? 

Also, can you add 1 new table and 2 new figures: 

**New Table**
Title:  Table 7.2 Baseline- Home Mouthwash Kit Return Rate
*This table will mimic Table 7.1 but will only have 4 columns:  Site,  Shipped (this would be total ever shipped);  Received; Total
*Can this data have a 1 week lag to account for lag time to get to the participant, complete collection and then ship the sample back to FNL? _Footnote: Shipped data are current through [date 1 week ago]._ 

**New Figures** 
Title: Figure 7.2 Baseline- Home Mouthwash Kit Return Rate
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/157617486/cdd61488-81dd-4fe8-9b36-54fbfc2f95d2)
Kits shipped would be the blue fill part and the kits received could be the line; left Y axis= # Kits Shipped; right Y axis Cumulative Return Rate; X axis = Date 
Footnote: Shipped data are current through [date 1 week ago].

Title: Figure 7.3 
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/157617486/f9301570-c3f7-4722-94cd-cedb81b65bfa)
Y axis=  Number of Kits Shipped
X axis = Site 
Coral bar= Total Kits Shipped
Turquoise bar= Total Kits Returned

**Comment by KELSEYDOWLING7** [comment]
@erincschwartz  Thank you for letting me know. 

@brotzmanmj  The counts all seem to be in order now. Please let me know if/when the team decides on how they want to lag Table 7.2 (with received lagged as well or not). For now I have not lagged either. I also recommend changing the column name from 'Shipped' to 'Ever Shipped' for clarification. Lastly, the 'Total' in that table really has no meaning, as we're counting those ever shipped (including received) as well as received. 

For Figure 7.2, please note that a second y axis is very hard to control. It will only take up a very small portion of the bottom right axis, so that line may never really look all that different. Meaning that I could have the second y axis from 0-30% or 0-50% but I can't get the ticks to stretch further up the right y axis, only increase the numbers on the tick line. It will not really be a proportional line. I think it worked significantly better in the operations report only because it was less than 10%. Here we have values of 30%+ but it looks the same as the 3% at first glance. Instead, I've made a slight adjusted to the request. Please let me know your thoughts on this.


![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/40759e35-1b5f-4d1a-a41e-403d54b82e9b)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/06e56691-e01a-4953-b0b8-111dec6f8791)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/8eb5cb1e-8b17-4cde-8328-fd867b5256dc)




![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/b74d4f10-a33f-4677-8f48-9e572b3185d7)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/2978ad88-0190-4e46-b71b-827d87571b17)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/ef006e8b-ba55-42a7-a148-ee7b72c4c8d5)
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/983579ee-d3fb-4abb-b512-5d6dd35e0399)


Please note their are 13 received kit that are excluded based on the criteria requested. Meaning BL D/T Clin Blood Collection or BL D/T Clin Urine Collection were collected before 4/1/24 and we've received their kits

**Comment by KELSEYDOWLING7** [comment]
@brotzmanmj  Updated Table 7.2 as requested
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/10f9b530-7125-4fdb-ada4-0179e7826956)

I also realized why the line in 7.2 looked much lower than we were expecting. The 24% was mapping to the number 24 on the y axis. Instead I made this second mountain the cumulative returns as a count, and put the percentage as the label.
![image](https://github.com/Analyticsphere/metricsReportsRequests/assets/101136647/260d5cd0-1d61-4584-badd-cee382fe8686)

**Comment by brotzmanmj** [comment]
This looks great, thanks

**Comment by KELSEYDOWLING7** [comment]
Great! Is it set to close the issue?

**Comment by brotzmanmj** [comment]
Yes thanks!
